### DISCUSSION TITLE
Debugging bare metal on mpfs-icicle-kit-es with GDB

### DISCUSSION THREAD

**Discussion by Fabien-Chouteau** [polarfire-soc/polarfire-soc-documentation#31] [Bare metal embedded software] [answered]
Hello,

Is there a way to get a GDB remote connection to the onboard FlashPro? For instance with openocd.

Regards,

**Comment by Cyril-Jean** [comment]
If you are one of the lucky people with a pre-production Icicle Kit, you need to use an external FlashPro JTAG probe. Then you can use openOCD.
You will be able to use the embedded FlashPro (USB connector J33) with the production boards coming at the end of the summer.

**Comment by Fabien-Chouteau** [comment]
Thank you @Cyril-Jean,

Where can I find an openOCD release and scripts that supports FlashPro and the mpfs-icicle-kit-es?

**Comment by Cyril-Jean** [comment, marked as answer]
Easiest is to the use SoftConsole v6.3. You can download the installer from the following links:
Linux host:http://soc.microsemi.com/download/reg/download.aspx?p=f%3dSoftConsole_v6_3_LIN
Windows host: http://soc.microsemi.com/download/reg/download.aspx?p=f%3dSoftConsole_v6_3_WIN

**Comment by Fabien-Chouteau** [comment]
It works, thank you @Cyril-Jean 

For the record I used this command line:
```shell
openocd/bin/openocd --command "set DEVICE MPFS" --file openocd/share/openocd/scripts/board/microsemi-riscv.cfg
```
